### PR TITLE
MWPW-114721 - Milo V2

### PR DESCRIPTION
**Pull request by auniverseaway**
**Note:** This is a breaking change for _non-live_ consumers.

* Locale support
* Template support
* Adobe Fonts support (including CJK)
* IMS & gnav prep
* Milo matches consumer API better
* Renamed nav to milonav
* Support for BYONav block
* Minor performance improvements
* Fixed issue with Helix V7 and LCP image

Resolves: [MWPW-114721](https://jira.corp.adobe.com/browse/MWPW-114721)

**Test URLs:**
* Before: https://main--milo--adobecom.hlx.page
* After: https://v2--milo--adobecom.hlx.page

**Consumer URLs:**
* Before: https://main--blog--adobe.hlx.page
* After: https://main--blog--adobe.hlx.page/?milolibs=v2
* Before: https://main--business-website--adobe.hlx.page
* After: https://main--business-website--adobe.hlx.page/?milolibs=v2

### CODE REVIEW

File: test/utils/utils.test.js [hunk 1]
```diff
@@ -71,19 +76,19 @@ describe('Utils', () => {
...
 
   it('Successfully dies parsing a bad config', () => {
     utils.parseEncodedConfig('error');
     expect(console.log.args[0][0].name).to.equal('InvalidCharacterError');
   });
 
-  it('Text getEnv()', () => {
-    expect(utils.getEnv()).to.equal('local');
-  });
+  // it('Test getEnv()', () => {
```
Comment (fullcolorcoder) [inline test/utils/utils.test.js:89]: getConfig instead now?

Comment (auniverseaway) [reply test/utils/utils.test.js:89]: Yup.

1. getEnv() should not be exported anymore.
2. The comments need to be removed, too.

Thank you!

File: libs/utils/utils.js [hunk 2]
```diff
@@ -75,25 +120,54 @@ export function loadStyle(href, callback) {
...
+      }
+      resolve();
+    })();
+  });
+  await Promise.all([styleLoaded, scriptLoaded]);
+}
+
 export async function loadBlock(block) {
   const { status } = block.dataset;
   if (!status === 'loaded') return block;
```
Comment (chrischrischris) [inline libs/utils/utils.js:152]: Can this ever be true?  `!status` will coerce the string into a bool which can not equal the `'loaded'` string.

Also I don't see where `block.dataset.status` is assigned `loaded`

File: libs/utils/utils.js [hunk 3]
```diff
@@ -109,13 +183,24 @@ export async function loadBlock(block) {
...
     blocks.splice(lcpIdx, 1);
-    await loader(lcpBlock, true);
+    await loadBlock(lcpBlock, true);
   }
+  const lcpImg = document.querySelector('main img');
+  return new Promise((resolve) => {
+    if (lcpImg && !lcpImg.complete) {
+      /* c8 ignore next 3 */
+      lcpImg.setAttribute('loading', 'eager');
+      lcpImg.addEventListener('load', () => resolve(lcpImg));
```
Comment (chrischrischris) [inline libs/utils/utils.js:198]: prob doesn't matter but should these event listeners be removed after load/error?

File: libs/utils/utils.js [hunk 4]
```diff
@@ -218,12 +303,20 @@ function decorateDefaults(el) {
...
 
 export function decorateNavs(el = document) {
-  const selectors = [];
-  if (getMetadata('nav') !== 'off') { selectors.push('header'); }
+  const selectors = ['header', 'footer'];
+  if (getMetadata('header') === 'off') {
+    document.body.classList.add('nav-off');
+    selectors.shift();
+  }
   if (getMetadata('footer') !== 'off') { selectors.push('footer'); }
```
Comment (chrischrischris) [inline libs/utils/utils.js:311]: if true, `selectors` will have footer twice.

File: libs/utils/utils.js (hunk 2 above)
Comment (auniverseaway) [reply libs/utils/utils.js:152]: > Also I don't see where block.dataset.status is assigned loaded

Good point. I think this came from boilerplate and we can remove.

File: libs/utils/utils.js (hunk 3 above)
Comment (auniverseaway) [reply libs/utils/utils.js:198]: That's _a lot_ (IMHO) of extra code to change. We'd have to make new functions to be able to remove them.

**Review by chrischrischris — APPROVED** [review]

**Review by seanchoi-dev — APPROVED** [review]

**Pull request merged**